File: app/src/main/java/com/cocobiz/app/domain/model/AppSettings.kt
package com.cocobiz.app.domain.model

data class AppSettings(
    val id: Long = 1L,
    val reminderDays: Int = 5,
    val notificationEnabled: Boolean = true,
    val emailEnabled: Boolean = false,
    val senderEmail: String = "",
    val darkMode: DarkModeOption = DarkModeOption.SYSTEM,
    val backupEnabled: Boolean = false,
    val defaultCycleDays: Int = 60
)

enum class DarkModeOption(val displayName: String) {
    LIGHT("Light"),
    DARK("Dark"),
    SYSTEM("System Default")
}

data class DashboardStats(
    val totalActiveSales: Int = 0,
    val totalCompletedSales: Int = 0,
    val totalRevenue: Double = 0.0,
    val upcomingReminders: Int = 0
)

File: backend/models/AppSettings.js
const mongoose = require('mongoose');

const appSettingsSchema = new mongoose.Schema({
  localId: { type: Number, default: 1, unique: true },
  reminderDays: { type: Number, default: 5 },
  notificationEnabled: { type: Boolean, default: true },
  emailEnabled: { type: Boolean, default: false },
  senderEmail: { type: String, default: '' },
  darkMode: { type: String, enum: ['LIGHT', 'DARK', 'SYSTEM'], default: 'SYSTEM' },
  backupEnabled: { type: Boolean, default: false },
  defaultCycleDays: { type: Number, default: 60 },
  updatedAt: { type: Number, default: Date.now }
});

module.exports = mongoose.model('AppSettings', appSettingsSchema);
